botocore/utils.py

```diff
@@ -2878,3 +3216,36 @@ def _get_global_endpoint(self, endpoint, endpoint_variant_tags=None):
             dns_suffix = self._DEFAULT_DNS_SUFFIX
 
         return f"https://{endpoint}.endpoint.events.{dns_suffix}/"
+
+
+def is_s3_accelerate_url(url):
+    """Does the URL match the S3 Accelerate endpoint scheme?
+
+    Virtual host naming style with bucket names in the netloc part of the URL
+    are not allowed by this function.
+    """
+    if url is None:
+        return False
+
+    # Accelerate is only valid for Amazon endpoints.
+    url_parts = urlsplit(url)
+    if not url_parts.netloc.endswith(
+        'amazonaws.com'
```

Comment:
Is accelerate only available in the default partition? I don't think this works for china or gov-cloud. If that was our previous behavior, that's fine but we should verify this isn't a regression.

Reply:
Transfer acceleration is not available in GovCloud: https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/govcloud-s3.html

I am not sure about China, but it is not mentioned here: https://www.amazonaws.cn/en/s3/features/

Reply:
And yes, this has been the existing behavior, encoded in `S3EndpointSetter`: https://github.com/boto/botocore/blob/762556d1745d42dfd56f4c2cdbe3e11ec417a4a7/botocore/utils.py#L2133-L2136